### ISSUE TITLE
[RyuJIT/arm64][JitStress=1] Assertion failed '(op2 == base) || (op2->gtEffectiveVal() == base)'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14206]
From a Windows x64_arm64_altjit run with `COMPlus_JitStress=1`:

Tests:
```
JIT\Directed\perffix\primitivevt\callconv3_il_d\callconv3_il_d.cmd
JIT\Directed\perffix\primitivevt\identity3_il_d\identity3_il_d.cmd
JIT\Methodical\VT\port\_il_dbghuge_gcref\_il_dbghuge_gcref.cmd
JIT\Methodical\VT\port\_rellcs_gcref\_rellcs_gcref.cmd
JIT\Methodical\VT\port\_dbglcs_gcref\_dbglcs_gcref.cmd
JIT\Methodical\unsafecsharp\_speed_relunsafe-3\_speed_relunsafe-3.cmd
JIT\Methodical\unsafecsharp\_speed_dbgunsafe-1\_speed_dbgunsafe-1.cmd
```

Example:
```
Assert failure(PID 11600 [0x00002d50], Thread: 21848 [0x5558]): Assertion failed '(op2 == base) || (op2->gtEffectiveVal() == base)' in 'DomainBoundILStubClass:IL_STUB_Array_Set(int,int,struct):this' (IL size 96)

    File: e:\gh\coreclr2\src\jit\gentree.cpp Line: 3836
    Image: E:\gh\coreclr2\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```

**Comment by BruceForstall** [comment]
@dotnet/arm64-contrib

**Comment by BruceForstall** [comment]
e.g., https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_arm64_altjit_checked_windows_nt_jitstress2_jitstressregs3/1/

**Comment by sandreenko** [comment]
Was fixed by #14914.
